### PR TITLE
keep compatibility with lease API

### PR DESCRIPTION
**Pull request by skeeey**
Signed-off-by: liuwei <liuweixa@redhat.com>

### CODE REVIEW

File: pkg/spoke/addon/lease_controller.go [hunk 1]
```diff
@@ -59,8 +59,11 @@ func NewManagedClusterAddOnLeaseController(clusterName string,
 		hubLeaseClient: hubLeaseClient,
 		leaseLister:    leaseInformer.Lister(),
 	}
+
+	// TODO because kube lease API is stable after kube v1.17, to keep compatibility, we does not
+	// register the lease informer here (if kube version is lower than v1.17, using lease informer
+	// will result in this controller failed), in the future, we will register lease informer here
```
Comment (qiujian16) [inline pkg/spoke/addon/lease_controller.go]: We do not add leaser informer to support kubernetes version lower than 1.17.  Lease v1 api is introduced in  v1.17, hence adding lease informer in this controller will cause the hang of informer cache sync and result in fatal exit of this controller. The code will be factored when we no longer support kubernetes version lower than 1.17.

Comment (skeeey) [reply pkg/spoke/addon/lease_controller.go]: updated

**Review by qiujian16 — COMMENTED** [review]
/approve
/lgtm

**Pull request merged**